compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/QueryDesugar.java

```diff
@@ -326,6 +324,25 @@ BLangStatementExpression desugar(BLangQueryExpr queryExpr, SymbolEnv env,
         return streamStmtExpr;
     }
 
+    private BType getMapType(BType type) {
+        BType refType = Types.getReferredType(type);
+        BType resultantType = symTable.mapType;
+        if (refType.tag == TypeTags.UNION) {
+            for (BType memberType : ((BUnionType) type).getMemberTypes()) {
+                BType resultType = getMapType(memberType);
+                if (resultType != symTable.mapType) {
+                    resultantType = resultType;
+                    break;
+                }
+            }
+        } else if (refType.tag == TypeTags.INTERSECTION) {
+            resultantType = getMapType(((BIntersectionType) refType).effectiveType);
```

Comment:
```suggestion
            return getMapType(((BIntersectionType) refType).effectiveType);
```

Reply:
I used this format to avoid codecov warnings for the last line in this method.

Reply:
This method returns `symTable.mapType` when I can't find a map type from the queryExpr. I am not quite sure whether that logic is needed since we are doing the invalid scenarios at typeChecker level. I couldn't find a code sample to trigger that logic too. I used this default case because similar kind of approach is used for table constructing scenario. For safe side we can keep this logic. WDYT @lasinicl @pcnfernando

Reply:
+1. Will update. Thanks.